### PR TITLE
Add system update caching script

### PR DESCRIPTION
**Pull request by ShinniUwU**
# Pull Request

## Description

This pull request introduces updates to `systemupdate.sh` to enhance its functionality by adding caching and improving its modularity. 

### Summary of Changes

- **Added Caching:** 
  - Implemented a caching mechanism to store update counts in `/tmp/system_update_status`. This reduces the number of times update checks need to be performed, improving performance.
  
- **Modularized Script:** 
  - Refactored the script into distinct functions for checking updates, reading cached results, and performing system upgrades. This makes the script more organized and easier to maintain.

- **Improved Handling:**
  - Ensured that the script correctly handles cases where the cache file may be deleted or missing, prompting it to recreate the cache when necessary.

### Motivation and Context

The updates aim to improve the performance of the update checking process and make the script more maintainable. By caching the results, the script avoids redundant operations, which is beneficial for users who frequently check for updates.

### Dependencies

- No new dependencies are introduced with this change.

### Related Issues
- There are no specific issues related to this pull request.
## Type of change

Please put an `x` in the boxes that apply:

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Documentation update** (non-breaking change; modified files are limited to the documentations)
- [ ] **Technical debt** (a code change that does not fix a bug or add a feature but makes something clearer for devs)
- [ ] **Other** (provide details below)

## Checklist

Please put an `x` in the boxes that apply:

- [x] I have read the [CONTRIBUTING](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My commit message follows the [commit guidelines](https://github.com/prasanthrangan/hyprdots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added a changelog entry.
- [ ] I have added necessary comments/documentation to my code.
- [ ] I have added tests to cover my changes.
- [ ] I have tested my code locally and it works as expected.
- [ ] All new and existing tests passed.

## Screenshots

(N/A)

## Additional context
The script has not been tested in a live environment. Please review the changes and test in your setup. If there are any issues or further refinements needed, I will be happy to address them.

### CODE REVIEW

**Comment by kRHYME7** [comment]
please do  a clean commitr

**Comment by ShinniUwU** [comment]
Could you elaborate on what you mean by cleaner commit?

**Comment by kRHYME7** [comment]
this

image

**Review by kRHYME7 — CHANGES_REQUESTED** [review]
The older implementation pops up the systemupdate terminal window to avoid window duplication.

**Review by kRHYME7 — CHANGES_REQUESTED** [review]
If you have changes in the waybar module, please add it here also. 

It pops up the terminal windows when updates are available. It's intrusive.

**Comment by kRHYME7** [comment]
We are too naive and wrote this script to be the shortest and lesser lines as possible, lol. We appreciate the refactoring for better maintainability.
I got some points here. 

>  This reduces the number of times update checks need to be performed, improving performance.


This part shouldn't be a problem, as waybar will only query this script once a day. Adding this cache file will still query the update in the background, giving more stress in the background. 
What I can recommend is to add SIG for this module so that users who want to re query can just right-click on the module to trigger the "SIGNAL" and rerun the script. Check waybar docs for this.

Your PR got some problems, see my comments for changes above.

File: Configs/.config/waybar/modules/updates.jsonc [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
+    "custom/pacman": {
         "format": "{}",
-        "rotate": ${r_deg},
-        "on-click": "hyprctl dispatch exec 'systemupdate.sh up'",
-        "interval": 86400, // once every day
-        "tooltip": true,
-        "signal": 20,
-    },
+        "interval": 86400, 
+        "exec": "~/.config/waybar/scripts/update-check.sh",
```
Comment (kRHYME7) [inline Configs/.config/waybar/modules/updates.jsonc:4]: why?

File: Configs/.config/waybar/modules/updates.jsonc [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
+        "exec-if": "exit 0", 
+        "on-click": "termite -e 'sudo pacman -Syu'; pkill -SIGRTMIN+8 waybar",
```
Comment (kRHYME7) [inline Configs/.config/waybar/modules/updates.jsonc:6]: this one is only from the example you can do something like ` systemupdate.sh`   on-click event

**Pull request closed without merging**